pkg/comp-functions/functions/vshnredis/backup.go

```diff
@@ -0,0 +1,374 @@
+						},
+						Key: "AWS_SECRET_ACCESS_KEY",
+					},
+				},
+			},
+			Backup: &k8upv1.BackupSchedule{
+				ScheduleCommon: &k8upv1.ScheduleCommon{
+					Schedule: k8upv1.ScheduleDefinition(comp.Spec.Parameters.Backup.Schedule),
+				},
+				BackupSpec: k8upv1.BackupSpec{
+					KeepJobs: pointer.Int(0),
+				},
+			},
+			Prune: &k8upv1.PruneSchedule{
+				ScheduleCommon: &k8upv1.ScheduleCommon{
+					Schedule: "@weekly-random",
+				},
+				PruneSpec: k8upv1.PruneSpec{
+					Retention: k8upv1.RetentionPolicy{
+						KeepLast:    retention.KeepLast,
+						KeepHourly:  retention.KeepHourly,
+						KeepDaily:   retention.KeepDaily,
+						KeepWeekly:  retention.KeepWeekly,
+						KeepMonthly: retention.KeepMonthly,
+						KeepYearly:  retention.KeepYearly,
+					},
+				},
+			},
+		},
+	}
+
+	return iof.Desired.PutIntoObject(ctx, schedule, comp.Name+"-backup-schedule")
+}
+
+func adjustHelmValues(ctx context.Context, comp *vshnv1.VSHNRedis, iof *runtime.Runtime) error {
+
+	l := controllerruntime.LoggerFrom(ctx)
+
+	release := &xhelm.Release{}
+
+	err := iof.Desired.Get(ctx, release, "release")
+	if err != nil {
+		return err
+	}
+
+	valueMap := map[string]any{}
+
+	err = yaml.Unmarshal(release.Spec.ForProvider.Values.Raw, &valueMap)
+	if err != nil {
+		return err
+	}
+
+	valueMap = addPVCAnnotation(valueMap)
+	valueMap = addPodAnnotation(valueMap)
+	valueMap = addBackupCM(valueMap)
+
+	l.V(1).Info("Final value map", "map", valueMap)
+
+	valueRaw, err := json.Marshal(valueMap)
+	if err != nil {
+		return err
+	}
+
+	release.Spec.ForProvider.Values.Raw = valueRaw
+
+	return iof.Desired.PutWithResourceName(ctx, release, "release")
+}
+
+func addPVCAnnotation(valueMap map[string]any) map[string]any {
+	persistenceMap := valueMap["master"].(map[string]any)["persistence"].(map[string]any)
+
+	persistenceMap["annotations"] = map[string]string{
+		"k8up.io/backup": "false",
+	}
+
+	valueMap["master"].(map[string]any)["persistence"] = persistenceMap
+	return valueMap
+}
+
+func addPodAnnotation(valueMap map[string]any) map[string]any {
+	masterMap := valueMap["master"].(map[string]any)
+
+	masterMap["podAnnotations"] = map[string]string{
+		"k8up.io/backupcommand":  "/scripts/backup.sh",
+		"k8up.io/file-extension": ".tar",
+	}
+
+	valueMap["master"] = masterMap
+	return valueMap
+
+}
+
+func addBackupCM(valueMap map[string]any) map[string]any {
+
+	masterMap := valueMap["master"].(map[string]any)
+
+	volumes := []corev1.Volume{
+		{
+			Name: backupScriptCMName,
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: backupScriptCMName,
+					},
+					DefaultMode: pointer.Int32(0774),
+				},
+			},
+		},
+	}
+
+	masterMap["extraVolumes"] = volumes
+
+	volumeMounts := []corev1.VolumeMount{
+		{
+			Name:      backupScriptCMName,
+			MountPath: "/scripts",
+		},
+	}
+
+	masterMap["extraVolumeMounts"] = volumeMounts
+
+	valueMap["master"] = masterMap
```

Comment:
All of these unchecked type casts and map access have the potential to panic. So as far as I can tell (potentially harmless looking) changes to the component can result in a panic here.

I can see that these should be fine in practice, but I'd still prefer this to be a bit more defensive and to check if the typecasts actually work and that the fields aren't nil